sormas-ui/src/main/java/de/symeda/sormas/ui/customexport/ExportConfigurationEditLayout.java

```diff
@@ -98,6 +95,44 @@ public ExportConfigurationEditLayout(
 		setComponentAlignment(buttonLayout, Alignment.MIDDLE_RIGHT);
 	}
 
+	private HorizontalLayout buildSelectionNameLayout() {
+
+		HorizontalLayout selectionNameLayout = new HorizontalLayout();
+		selectionNameLayout.setMargin(false);
+
+		tfName = new TextField(I18nProperties.getPrefixCaption(ExportConfigurationDto.I18N_PREFIX, Captions.ExportConfiguration_NAME));
+		tfName.setWidth(350, Unit.PIXELS);
+		tfName.setRequiredIndicatorVisible(true);
+		if (this.exportConfiguration.getName() != null) {
+			tfName.setValue(this.exportConfiguration.getName());
+		}
+		selectionNameLayout.addComponent(tfName);
+
+		checkBoxPublicExport = new CheckBox();//(I18nProperties.getPrefixCaption(ExportConfigurationDto.I18N_PREFIX, Captions.ExportConfiguration_reportsToPublic));
+		checkBoxPublicExport.setWidth(350, Unit.PIXELS);
+		checkBoxPublicExport.setValue(this.exportConfiguration.isReportsToPublic());
+		checkBoxPublicExport.setEnabled(hasRights());
+
+		Label publicExportLabel = new Label(I18nProperties.getPrefixCaption(ExportConfigurationDto.I18N_PREFIX, Captions.ExportConfiguration_reportsToPublic));
+		CssStyles.style(publicExportLabel, CssStyles.H5);
+		CssStyles.style(publicExportLabel, CssStyles.LABEL_LARGE);
+
+		HorizontalLayout selectionPublicExportLayout = new HorizontalLayout();
+		selectionPublicExportLayout.setMargin(false);
+
+		selectionPublicExportLayout.addComponent(publicExportLabel);
+		selectionPublicExportLayout.addComponent(checkBoxPublicExport);
+
+		selectionNameLayout.addComponent(selectionPublicExportLayout);
+
+		return selectionNameLayout;
+	}
+
+	private boolean hasRights() {
```

Comment:
please give this a better name, hasRIghts is too generic but this method checks only the right for managing public exports

Reply:
Also this method is called too many times and it has impact on the performance as UserProvider.getCurrent().hasUserRight accesses the backend and also the DB at some point.

It is enough to call it once in the constructor of the component.